src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesTestComponents.scala

```diff
@@ -102,14 +102,16 @@ private[spark] object SparkAppLauncher extends Logging {
       sparkHomeDir: Path): Unit = {
     val sparkSubmitExecutable = sparkHomeDir.resolve(Paths.get("bin", "spark-submit"))
     logInfo(s"Launching a spark app with arguments $appArguments and conf $appConf")
-    val commandLine = Array(sparkSubmitExecutable.toFile.getAbsolutePath,
+    val commandLine = mutable.ArrayBuffer(sparkSubmitExecutable.toFile.getAbsolutePath,
       "--deploy-mode", "cluster",
       "--class", appArguments.mainClass,
       "--master", appConf.get("spark.master")
     ) ++ appConf.toStringArray :+
-      appArguments.mainAppResource :+
-      appArguments.appArgs.mkString(" ")
+      appArguments.mainAppResource
+    if (appArguments.appArgs.nonEmpty) {
+      commandLine += appArguments.appArgs.mkString(" ")
```

Comment:
This renders the logic of the driverArgs match in KubernetesClientApplication useless. As such, we should take that out (unless you think is should be left in for sanity check)

Reply:
This error is why I added that match statement in the first place.

Reply:
Think we should allow for empty string arguments to be passed if that's really what the user intends to do. There's a difference between an empty array of arguments and a non-empty array of empty arguments. These shouldn't necessarily be treated equally.